api-gateway/src/main/java/com/petclinic/bffapigateway/domainclientlayer/InventoryServiceClient.java

```diff
@@ -323,7 +323,7 @@ public Mono<ProductResponseDTO> consumeProduct(String inventoryId, String produc
                 .uri(uriBuilder.buildAndExpand(inventoryId, productId).toUri())
                 .retrieve()
                 .onStatus(HttpStatusCode::is4xxClientError,
-                        resp -> Mono.error(new NotFoundException("Product not found in inventory: " + inventoryId)))
+                        resp -> Mono.error(new InventoryNotFoundException("Product not found in inventory: " + inventoryId, NOT_FOUND)))
```

Comment:
Maybe change the exception's name to ProductNotFoundException as it's a product not found in inventory but the rest of the tests look great!